Add docs for migrating jobs to EKS cluster

xmudrii opened the pull request:
As discussed previously and recently at the SIG K8s infra meeting held on 2023-06-21, we want to add some docs for migrating jobs to EKS cluster. The purpose of docs is to clarify some important points about migration, such as:

- determining and monitoring resource requests and limits
- known issues
- community ownership of external resources

This document is based on instructions written by @BenTheElder, @dims, @rjsadow, and everyone involved with #29722

/assign @ameukam @dims @rjsadow @iftachk @jeefy 
/hold
for discussion

xmudrii commented:
/test pull-test-infra-unit-test

rjsadow commented:
This is a great writeup! There are two pieces of information that I think should be added to clarify which jobs should be excluded from migrating at this time. 

1. Jobs that have services that don't 1-to-1 translate to EKS. Boskos being one where if used on GCP shouldn't migrate to EKS. [Slack ref](https://kubernetes.slack.com/archives/C09QZ4DQB/p1687239371001069).
2. You touch on this a bit in `Community Ownership of Resources` but jobs that have or rely on any credentials or secrets in the default cluster can't be moved right now.

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+similar to our GCP/GKE based clusters, but there are some significant
+differences such as:
+
+- Operating system on worker nodes where we run jobs is Ubuntu 22.04
```
Actually, it's 20.04 (EKS optimised)
<img width="507" alt="Screenshot 2023-06-30 at 09 12 39" src="https://github.com/kubernetes/test-infra/assets/9121459/bcde2eb7-0dd5-4a0c-a835-4e558e14fec1">

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+  jobs are defined in the [kubernetes/test-infra][k-test-infra] repository. You
+  can use GitHub or our [Code Search tool][cs-test-infra] to find the file
+  where this job is defined (search by job name).
+- Look for a `cluster:` key in the job definition. If there isn't one or it's
```
It might be helpful to point out that it's part of spec for cases where it's not defined.
```suggestion
- Look for a `spec.cluster` key in the job definition.
```

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+    to the `cluster:` key)
+- Save the file, commit the change, create a branch and file a PR
+- Once the PR is merged, follow the guidelines in the Post Migration Period
+  section of this document to ensure that the job remains stable
```
I would add here to check job definition for images referencing docker.io. Last week we had a problem with rate limiting. It's not strictly related to EKS migration but we still should encourage people to replace them with ECR mirrors from [the official gallery](https://gallery.ecr.aws/).

xmudrii replied:
@dims mentioned that we might want to first sync with AWS folks before doing this on large scale, so I'd leave it from recommendations until then.

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+Jobs running in community-owned clusters must have resource requests and 
+limits specified for cluster-autoscaler to work properly. However, determining
+this is not always easy and incorrect requests and limits can cause the job
+to start flaking or failing.
```
Or even affect other jobs running in the cluster in case it causes CPU throttling.

xmudrii replied:
AFAIK, throttling only applies to a specific cgroupns and each pod is supposed to have its own namespace. That being said, hopefully, other jobs shouldn't be affected by throttling.

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+
+## Known Issues
+
+- Go doesn't respect cgroups, i.e. CPU limits are not going to be respected by
```
I might exaggerate here, but using full name will be less ambiguous. 
```suggestion
- Golang/Go doesn't respect cgroups, i.e. CPU limits are not going to be respected by
```

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+- Go doesn't respect cgroups, i.e. CPU limits are not going to be respected by
+  Go applications
+  - This means that a Go application will try to use all available CPUs on
+    node, even though it's limited, e.g. to 2
```
```suggestion
    node, even though it's limited by k8s resource limit, e.g. to 2
```

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+    issue by setting `GOMAXPROCS` environment variable for that job to the
+    value of `cpu` limit. There are also ways to automatically determine
+    `GOMAXPROCS`, such as [`automaxprocs`][automaxprocs]
+- Kernel configs are not available inside the job's test pod so kubeadm can
```
`can` or `cannot`?
```suggestion
- Kernel configs are not available inside the job's test pod so kubeadm cannot
```

xmudrii replied:
Replaced with `might`.

pkprzekwas commented on docs/eks-jobs-migration.md:
```diff
@@ -0,0 +1,216 @@
+- Once the PR is merged, follow the guidelines in the Post Migration Period
+  section of this document to ensure that the job remains stable
+
+If you have any trouble, please leave a note in the following
+[GitHub issue][test-infra-gh-issue] or come to
+[#sig-k8s-infra][slack-k8s-infra] or [#sig-testing][slack-sig-testing] Slack
+channel to ask for help.
```
This block is duplicated. I'm ok with that, but we could also redirect to the bottom part of this document.

xmudrii commented:
I addressed most of comments from @rjsadow, @iftachk, and @pkprzekwas. I'd like to further elaborate on this comment from @rjsadow:

> You touch on this a bit in Community Ownership of Resources but jobs that have or rely on any credentials or secrets in the default cluster can't be moved right now.

The point of `Community Ownership of Resources` is to enable and facilitate moving jobs that rely on any credentials or secrets in the default cluster. We want to, eventually, move those jobs as well. They can be moved, but that requires going through steps described in the `Community Ownership of Resources` section. That's also stated in `How To Migrate Jobs?`:

> **IMPORTANT: if you see any entries under `label` or `volumeMounts` that might indicate that a job is using some external non-GCP resource (e.g. credentials for some cloud platform), you need to check if the Community Ownership criteria is satisfied for that resource (see the Community Ownership section for more details)**

As long as someone is willing to work with us to satisfy at least the minimum criteria, we shouldn't block affected jobs from being moved to the new cluster.

ameukam commented:
/lgtm

rjsadow commented:
/lgtm

xmudrii commented:
@ameukam I think this is in a good state to get merged. Do you want to approve it?

ameukam commented:
/approve

ameukam commented:
/hold cancel
